blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/BluefloodServiceStarter.java

```diff
@@ -313,4 +316,20 @@ public void run() {
         startEventListenerModules();
         log.info("All blueflood services started");
     }
+
+    @VisibleForTesting
+    public static boolean SkipInstantiateRestartGauge = false;
+
+    @VisibleForTesting
+    public static boolean ThrowInsteadOfExit = false;
+
+    private static void SystemExit(int status) throws BluefloodServiceStarterException {
```

Comment:
Method should start with small letter. Also does this method have to be static? This method seems a little redundant to me especially if we get rid of this flag ThrowInsteadOfExit. 

I would have taken this approach if we really want to get rid of System.exit calls. We should throw exception everwhere we have System.exit(1) calls. Store the exit code inside the exception. Catch the exception in the main method and exit with proper exit code.